wolfssl/ssl.h

```diff
@@ -585,7 +585,9 @@ struct WOLFSSL_X509_STORE {
     int             refCount;         /* reference count */
 };
 
-#define WOLFSSL_NO_WILDCARDS   0x4
+#define WOLFSSL_ALWAYS_CHECK_SUBJECT 0x1
+#define WOLFSSL_NO_WILDCARDS         0x2
+#define WOLFSSL_NO_PARTIAL_WILDCARDS 0x4
```

Comment:
`WOLFSSL_NO_WILDCARDS` is mapped to `X509_CHECK_FLAG_NO_WILDCARDS`, which OpenSSL defines as 0x2.  I added `WOLFSSL_NO_PARTIAL_WILDCARDS` in this PR, which OpenSSL defines as 0x4.  Switching these to be accurate compatibility values seemed like the best solution path, and with minimal impact since applications should be using the define vs actual values.